### DISCUSSION TITLE
Dexcom G7 in companion mode synching with huge delay

### DISCUSSION THREAD

**Discussion by guinnberg** [NightscoutFoundation/xDrip#3148] [Q&A]
Hi,

I just discovered today that I can use the Companion mode so I don't need internet, which is great especially when I'm flying in a plane and I use xDrip mainly for my Garmin watch.

The problem I have is that it takes around 4 minutes for xDrip to get the updated value from the Dexcom app, so that means that when I see a value in xDrip, it's already pretty much outdated.

I activated the bluetooth companion mode, but I don't see any difference. Is this something that it'll fix itself over time or it'll fix when I insert a new sensor?

TIA

**Comment by jamorham** [comment]
I know about this issue and a fix is being worked on but its a little complicated to get right which is why it hasn't happened yet.

**Reply by guinnberg** [reply]
I understand it can be complex, but just having a button to reset the time when it should check it'd be cool, I wouldn't mind having to press the thing once in a while.

This been said, it looks like it fixed itself... I get the xDrip notification as "now" and there's like a minute difference between current time and Dexcom reading

**Reply by ah7675** [reply]
Try enabling Bluetooth companion.

**Reply by guinnberg** [reply]
I already did

**Reply by ah7675** [reply]
> I already did

Oh, sorry I didn't notice that you have mentioned in last paragraph.

How about disable "Quick at glance" in G7 app for 10 minutes and then re-enable it?

**Reply by guinnberg** [reply]
Now this morning it works real time, so I guess it needed some time to synch properly.

Thanks for your help!

**Comment by jamorham** [comment]
I know you've had this issue spontaneously resolve but I put some fixes in to try to ensure it can resolve much quicker in future so I would appreciate if you could test the latest xDrip nightly to ensure that this is still working properly for you.

**Reply by guinnberg** [reply]
Sorry, I just saw this now... Do you still need me to test it? And if so, what would be the best method? Should I also activate the Bluetooth companion mode?

**Reply by Navid200** [reply]
You said your problem was resolved, right?  

Use the latest Nightly: https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.10.24
Is it still working fine after you update?
